### ISSUE TITLE
Mosquitto broker Addon  - Duplicate log_timestamp_format value in configuration

### ISSUE THREAD

**Issue by thomas-amler-idnow** [home-assistant/addons#2640]
### Describe the issue you are experiencing

Since Update 6.1.3 for the Mosquitto broker addon I get a duplicate config key error for "log_timestamp_format" config parameter.
Mosquitto can not start anymore due to this.

### What type of installation are you running?

Home Assistant Supervised

### Which operating system are you running on?

Home Assistant Operating System

### Which add-on are you reporting an issue with?

Mosquitto broker

### What is the version of the add-on?

6.1.3

### Steps to reproduce the issue

1. Update to Mosquitto broker 6.1.3
2. Start the Mosquitto broker addon
3. Refresh Mosquitto broker Addon status page
4. Look at the logs, see below

### Anything in the add-on logs that might be useful for us?

```txt
Addon logs show duplicate config key "log_timestamp_format":

s6-rc: info: service s6rc-oneshot-runner: starting
s6-rc: info: service s6rc-oneshot-runner successfully started
s6-rc: info: service fix-attrs: starting
s6-rc: info: service fix-attrs successfully started
s6-rc: info: service legacy-cont-init: starting
cont-init: info: running /etc/cont-init.d/mosquitto.sh
[15:21:37] INFO: SSL is not enabled
cont-init: info: /etc/cont-init.d/mosquitto.sh exited 0
cont-init: info: running /etc/cont-init.d/nginx.sh
cont-init: info: /etc/cont-init.d/nginx.sh exited 0
s6-rc: info: service legacy-cont-init successfully started
s6-rc: info: service legacy-services: starting
services-up: info: copying legacy longrun mosquitto (no readiness notification)
services-up: info: copying legacy longrun nginx (no readiness notification)
s6-rc: info: service legacy-services successfully started
[15:21:37] INFO: Starting NGINX for authentication handling...
[15:21:38] INFO: Starting mosquitto MQTT broker...
1661865698: Loading config file /share/mosquitto/logging.conf
1661865698: Error: Duplicate log_timestamp_format value in configuration.
1661865698: Error found at /share/mosquitto/logging.conf:1.
1661865698: Error found at /etc/mosquitto/mosquitto.conf:37.
[13:21:38] WARNING: Halt add-on
s6-rc: info: service legacy-services: stopping
[13:21:38] INFO: Service restart after closing
s6-rc: info: service legacy-services successfully stopped
s6-rc: info: service legacy-cont-init: stopping
s6-rc: info: service legacy-cont-init successfully stopped
s6-rc: info: service fix-attrs: stopping
s6-rc: info: service fix-attrs successfully stopped
s6-rc: info: service s6rc-oneshot-runner: stopping
s6-rc: info: service s6rc-oneshot-runner successfully stopped
```

**Comment by mdegat01** [comment]
Looks like you're using the `customize` option and have specified `log_timestamp_format` in your config customization. You need to remove yours as this addon now specifies that value.

**Comment by tomlut** [comment]
I'm seeing the same error:
```
1661894756: Error: Duplicate log_timestamp_format value in configuration.
1661894756: Error found at /share/mosquitto/mosquitto.conf:2.
1661894756: Error found at /etc/mosquitto/mosquitto.conf:37.
[21:25:56] WARNING: Halt add-on
```

I am not specifying the `log_timestamp_format`:
```
logins: []
require_certificate: false
certfile: fullchain.pem
keyfile: privkey.pem
customize:
  active: true
  folder: mosquitto
```

**Comment by cabletie** [comment]
Would have been nice to know this was a breaking change. It's not nice to have your MQTT broker suddenly stop working. Maybe a better solution is to still enable the log_timestamp_format config item as an override...

**Comment by mdegat01** [comment]
@cabletie the `customize` option allows you to inject your own config into the mosquitto config file. If it works for you I'm glad but it it is literally impossible for us to predict breaking changes for users that use that option. The developers of the addon have literally no idea how your broker is configured, what is or isn't in use, what works and what doesn't, etc.

You're welcome to use it but you do so at your own risk. You should consider every update to the addon a potential breaking change if you do. If you make extensive use of it I would recommend instead cloning the addon and making your own with a config that you control. Or hosting your own broker externally.

**Comment by ecsfang** [comment]
Had the same problem, and just as @tomlut said ("I don't remember creating that but I must have at some stage"), I don't either recall that I had added the config file with a single entry of a timestamp definition, but maybe I have ...

I removed the definition from /share/mosquitto/mosquitto.conf and all is well now. 

But, if I really wanted to customize the timestamp format, how would I do that if I can't add it using the customize option ... ?

**Comment by thomas-amler-idnow** [comment]
@mdegat01 Unfortunately I don't have this option in the customizations.

My config:
```
logins: []
require_certificate: false
certfile: fullchain.pem
keyfile: privkey.pem
customize:
  active: true
  folder: mosquitto
anonymous: false
```

**Comment by ecsfang** [comment]
@mdegat01: with `customize` enabled, the file "/share/mosquitto/mosquitto.conf" will be used. In my case, that file contained the timestamp definition, and if I just commented it out (ie. in the file "/share/mosquitto/mosquitto.conf") it worked fine again.
I guess you could do the same by disabling (or remove) the `customize` settings in your configuration (I have not tried that).

**Comment by thomas-amler-idnow** [comment]
@ecsfang The only file I have in /share/mosquitto/ is a logging.conf. It contained the log_timestamp_format parameter which I already commented as a workaround.
I actually thought this file has been put there by the update. But maybe I created it myself in the past. :thinking: I don't remember, but probably it was me.
Thanks for pointing it out. I should have guessed that from the different file locations as well. :grinning:
